### PR TITLE
Fix calculation of revenue on revenue reports

### PR DESCRIPTION
**Pull request by mayankagnihotri7**
### **What :**
[Notion card](https://www.notion.so/saeloun/Fix-the-calculations-on-the-revenue-report-f09523067da84bd38a5a829dc61733a2)

### **Why :**
- To fix the issue in production where the unpaid and paid amount displayed is incorrect. The calculations for the total unpaid amount are wrong.

### **Preview :**
**Screenshots:**
screencapture-localhost-3000-reports-revenue-by-client-2023-02-08-15_21_34

### CODE REVIEW

**Comment by vipulnsward** [comment]
@mayankagnihotri7 is this ready for review?

**Comment by mayankagnihotri7** [comment]
@vipulnsward Yes it is ready for review. I'll assign Shalaka for the review.

**Comment by shalapatil** [comment]
@mayankagnihotri7 Please rename unpaid amount to outstanding in FE as well as backend.  Unpaid word makes confusion now as we have separated overdue from it.

**Comment by mayankagnihotri7** [comment]
@shalapatil I've made the requested changes. Please review.

File: app/models/client.rb [hunk 1]
```diff
@@ -112,12 +112,13 @@ def register_on_stripe!
...
 
   def payment_summary(duration)
-    status_and_amount = invoices.during(duration).group(:status).sum(:amount)
+    status_and_amount = invoices.kept.during(duration).group(:status).sum(:amount)
     status_and_amount.default = 0
     {
       name:,
       paid_amount: status_and_amount["paid"],
-      unpaid_amount: status_and_amount["sent"] + status_and_amount["viewed"] + status_and_amount["overdue"]
+      unpaid_amount: status_and_amount["sent"] + status_and_amount["viewed"],
```
Comment (shalapatil) [inline app/models/client.rb:120]: change this to outstanding

File: app/views/internal_api/v1/reports/client_revenues/index.json.jbuilder [hunk 2]
```diff
@@ -6,11 +6,12 @@ json.currency current_company.base_currency
...
-  json.unpaid_amount client[:unpaid_amount]
-  json.total_amount client[:paid_amount] + client[:unpaid_amount]
+  json.outstanding_amount client[:unpaid_amount]
+  json.overdue_amount client[:overdue_amount]
+  json.total_amount client[:paid_amount] + client[:unpaid_amount] + client[:overdue_amount]
 end
 json.summary do
   json.total_paid_amount summary[:total_paid_amount]
-  json.total_unpaid_amount summary[:total_unpaid_amount]
+  json.total_outstanding_amount summary[:total_unpaid_amount]
```
Comment (shalapatil) [inline app/views/internal_api/v1/reports/client_revenues/index.json.jbuilder:15]: fix this

**Review by shalapatil — COMMENTED** [review]
Please rename at all the places.

**Comment by mayankagnihotri7** [comment]
@shalapatil I've made the requested changes. Please review.

**Review by shalapatil — COMMENTED** [review]
LGTM

**Comment by mayankagnihotri7** [comment]
@shalapatil Could you please approve the PR as well since you gave LGTM?

**Review by shalapatil — APPROVED** [review]

**Pull request merged**